Makefile

```diff
@@ -0,0 +1,90 @@
+# Copyright 2025 The Kubeflow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Setting SHELL to bash allows bash commands to be executed by recipes.
+# Options are set to exit when a recipe line exits non-zero or a piped command fails.
+SHELL = /usr/bin/env bash -o pipefail
+.SHELLFLAGS = -ec
+
+# run the whole recipe in one shell (so one cd covers all lines)
+.ONESHELL:
+.PHONY: verify
+
+
+PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
+
+PY_DIR := $(PROJECT_DIR)/python
+
+REPO := github.com/kubeflow/sdk
+# Location to install tool binaries
+LOCALBIN ?= $(PROJECT_DIR)/bin
+
+# Tool versions
+UV_VERSION=0.7.2
+RUFF_VERSION=0.12.3
```

Comment:
Is that important which version of uv and ruff we run ?
We can consider to place them under [`bin/` if specific version is required](https://github.com/kubeflow/trainer/blob/master/Makefile#L18).